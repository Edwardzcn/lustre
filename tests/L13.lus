node nats () returns (n : int)
let
  n = 0 -> 1 + pre n;
tel

node id (from : int) returns (n : int)
let
  n = from;
tel


node main(c : bool) returns (n : int)
let
  n = condact(c,id(nats()),42);
tel




