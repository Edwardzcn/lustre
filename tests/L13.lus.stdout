node (c : bool)
  returns (n)
  assumes ()
  shows ()
let
  valof_pre___no_static = pre __no_static$1
  __no_static$1 = 0 -> Add(1, valof_pre___no_static)
  from = __no_static$1 when c
  __no_static = from
  valof_current = current __no_static
  valof_pre_n = pre n
  valof_arr = 42 -> valof_pre_n
  n = ITE(c, valof_current, valof_arr)
tel
--- Step 1 ---
c : bool  = False
n = 42
--- Step 2 ---
c : bool  = False
n = 42
--- Step 3 ---
c : bool  = True
n = 2
--- Step 4 ---
c : bool  = False
n = 2
--- Step 5 ---
c : bool  = False
n = 2
--- Step 6 ---
c : bool  = True
n = 5
--- Step 7 ---
c : bool  = True
n = 6
--- Step 8 ---
c : bool  = False
n = 6
--- Step 9 ---
c : bool  = False
n = 6
--- Step 10 ---
c : bool  = (EOF)
